frontend/components/cards/ModuleUnionCard.tsx

```diff
@@ -8,6 +9,25 @@ interface ModuleUnionCardProps {
   selected?: boolean;
 }
 
+interface HighlightEllipsisProps {
+  unsafeHTML: string;
+}
+
+const HighlightEllipsis: FC<HighlightEllipsisProps> = ({ unsafeHTML }) => (
+  <Box
+    sx={{
+      "& *": { fontSize: "0.8rem", backgroundColor: "rgba(0, 0, 0, 0)" },
+      "& mark": { fontWeight: "500", color: "white" },
+    }}
+  >
+    <HTMLEllipsis
+      unsafeHTML={unsafeHTML.replace(/(<(h[1-6])>.*?<\/h[1-6]>)/gi, "")}
```

Comment:
Elasticsearch sometimes gives us hightlights content that contains empty `p` tags, i.e., `<p></p>`.
So, maybe we should add another regex replacement to get rid of those. (I looked briefly in the elasticsearch documentation and didn't see a way to avoid this from the backend.)

The additional rule for remove empty tags could be `.replace(/<(p|div)><\/(p|div)>/gi, "")` (May as well remove both empty `p` tags and empty `div` tags, although I haven't encountered empty divs yet.)